dev/src/generate_sql_schema.py

```diff
@@ -693,6 +700,30 @@ class Helper:
         END;
         $notify_trigger$ LANGUAGE plpgsql;
 
+        CREATE FUNCTION log_modified_related_models() RETURNS trigger AS $log_modified_related_trigger$
+        DECLARE
+            operation TEXT;
+            fqid TEXT;
+            ref_column TEXT;
+            foreign_table TEXT;
+            foreign_id TEXT;
+        BEGIN
+            operation:= LOWER(TG_OP);
+            ref_column := TG_ARGV[1];
+            foreign_table := TG_ARGV[0];
+
+            EXECUTE format('SELECT $1.%s', ref_column) INTO foreign_id USING NEW;
+            IF (TG_OP = 'DELETE') THEN
+                EXECUTE format('SELECT $1.%s', ref_column) INTO foreign_id USING OLD;
+            END IF;
+
+            fqid := foreign_table || '/' || foreign_id;
```

Comment:
The `foreign_id` is NULL if a collection is inserted or updated and a not required field is empty, e.g. insert a user with username only (gender_id is empty). The trigger is executed and inserts a row into os_notify_log_t with an empty fqid.

Reply:
I also run into this problem. I did not see this comment and it took me some time to debug it.

Please change the os_notify_log_t table (line 105) to:
```sql
CREATE TABLE os_notify_log_t (
    id integer PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
    operation varchar(32),
    fqid varchar(256) NOT NULL,
    xact_id xid8,
    timestamp timestamptz
);
``` 

(add the `NOT NULL` constain to the fqid column).
This prefents postgres to add invalid data to the os_notify_log_t table